lib/EditableList/EditableListForm.js

```diff
@@ -221,6 +227,20 @@ class EditableListForm extends React.Component {
     return null;
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.props.shouldReinitialize && this.shouldUpdateStatus(prevProps.initialValues)) {
+      this.setState({ status: buildStatusArray(this.props.initialValues.items) });
+    }
+  }
+
+  shouldUpdateStatus(previousValues) {
+    return (
+      this.props.initialValues &&
+      previousValues &&
+      this.props.initialValues !== previousValues
```

Comment:
@artem-blazhko `intialValues` is an object with some nested structure (an array) so `!==` won't work here. You will need to use a deep comparison with https://lodash.com/docs/#isEqual for example.

Reply:
@mkuklis Fixed